### PR TITLE
Range-ify std.path.buildNormalizedPath by adding toNormalizedPath()

### PR DESCRIPTION
**Pull request by WalterBright**
http://dlang.org/phobos/std_path.html#.buildNormalizedPath

buildNormalizedPath is rewritten to use toNormalizedPath internally. Most of the unit tests were moved to toNormalizedPath.

100% unit test coverage

### CODE REVIEW

**Comment by WalterBright** [comment]
I'm not terribly happy with this as it does not return a random access range. That makes it less composable than I'd like. But I'll leave that for another time.

File: std/path.d [hunk 1]
```diff
@@ -1823,138 +1615,414 @@ unittest
...
+        @property C front()
+        {
+            return c;
+        }
+
+        void popFront()
+        {
+            C lastc = c;
+            c = c.init;
+            if (element.empty)
```
Comment (andralex) [inline std/path.d]:
```d
if (!element.empty)
{
    c = getElement0();
    return;
}
... main logic goes here ...
```

Vastly easier to read and reason about.

File: std/path.d [hunk 2]
```diff
@@ -1823,138 +1615,414 @@ unittest
...
+                        if (isDot(e))
+                            continue;
+                        if (isDotDot(e))
+                        {
+                            --n;
+                            if (n == 0)
+                            {
+                                elements = elements2;
+                                element = element[0..0];
+                                goto L1;
```
Comment (andralex) [inline std/path.d]: `continue L1;`

File: std/path.d [hunk 3]
```diff
@@ -1823,138 +1615,414 @@ unittest
...
+                        }
+                        else
+                            ++n;
+                    }
+                }
+
+                if (lastc == dirSeparator[0] || lastc == lastc.init)
+                    c = getElement0();
+                else
+                    c = dirSeparator[0];
```
Comment (andralex) [inline std/path.d]: `static assert(dirSeparator.length == 1);` otherwise the whole approach needs rethinking.

**Comment by andralex** [comment]
lgtm modulo these nits

File: std/path.d (hunk 2 above)
Comment (WalterBright) [inline std/path.d]: that doesn't compile, because it's not inside L1's loop

**Pull request merged**